tensorbay/opendataset/_utility/nuScenes.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python3
+#
+# Copyright 2021 Graviti. Licensed under MIT License.
+#
+# pylint: disable=invalid-name
+
+"""nuScenes method for open dataset."""
+import json
+import os
+from bisect import bisect
+from itertools import accumulate, product
+from typing import Any, Dict, List
+
+
+def get_info_with_token(info_path: str, annotation_part: str) -> Dict[str, Any]:
+    """Parse the coco-like label files.
+
+    Arguments:
+        info_path: The Path of json files.
+        annotation_part: The name of json file.
+
+    Returns:
+        A annotation dict with token key.
+
+    """
+    filepath = os.path.join(info_path, f"{annotation_part}.json")
+    with open(filepath, "r") as fp:
+        info = json.load(fp)
+    return {item.pop("token"): item for item in info}
+
+
+def get_info_with_determined_token(
+    info_path: str,
+    annotation_part: str,
+    determined_token: str = "sample_token",
+    no_key_frame: bool = False,
+) -> Dict[str, List[Any]]:
+    """Parse the coco-like label files.
+
+    Arguments:
+        info_path: The Path of json files.
+        annotation_part: The name of json file.
+        determined_token: The key name of the dict.
+        no_key_frame: whether need key frame.
+
+    Returns:
+        A annotation dict with determined token key.
+
+    """
+    filepath = os.path.join(info_path, f"{annotation_part}.json")
+    with open(filepath, "r") as fp:
+        info = json.load(fp)
+    info_with_keys: Dict[str, List[Any]] = {}
+    for item in info:
+        if no_key_frame or item["is_key_frame"]:
+            info_with_keys.setdefault(item.pop(determined_token), []).append(item)
+    return info_with_keys
+
+
+def uncompress_rle(original_code: str, height: int, width: int) -> List[int]:
+    """Uncompress function for RLE of coco API.
+
+        Source Code: https://github.com/cocodataset/cocoapi/blob/master/common/maskApi.c
+        Func: rleFrString
+
+    Arguments:
+        original_code: The original ASCII code.
+        height: The height of mask.
+        width: The width of mask.
+
+    Returns:
+        RLE which need to be transposed.
+
+    """
+    pointer = 0
+    rle: List[int] = []
+    m = 0
+    while pointer < len(original_code):
```

Comment:
use `for` + `next` to replace `while` + `index`?

Reply:
` rle: List[int] = []
    m = 0
    code_generator = iter(original_code)
    for code in code_generator:
        x = 0
        k = 0
        more = 1
        while more:
            char = ord(code) - 48
            x |= (char & 0x1F) << 5 * k
            more = char & 0x20
            code = next(code_generator)
            k += 1
            if not more and (char & 0x10):
                x |= -1 << 5 * k
        if m > 2:
            x += rle[m - 2]
        rle.append(x)
        m += 1`
This code is wrong, because the `code` will `next` once more in the next loop.